libsignal-protocol/src/crypto/openssl.rs

```diff
@@ -0,0 +1,185 @@
+use crate::{
+    crypto::{Crypto, Sha256Hmac, Sha512Digest, SignalCipherType},
+    errors::{FromInternalErrorCode, InternalError, IntoInternalErrorCode},
+};
+use openssl::{
+    hash::{Hasher, MessageDigest},
+    nid::Nid,
+    symm::{Cipher, Crypter, Mode},
+};
+
+pub struct OpenSSLCrypto;
```

Comment:
that also removes a lot of things related to synchronization stuff like `Mutex`s and `lock`ing which also is nice thing to see here.